server/jwt_test.go

```diff
@@ -3836,3 +3836,194 @@ func TestJWTNoOperatorMode(t *testing.T) {
+	aJwt1, err := claim.Encode(oKp)
+	require_NoError(t, err)
+	claim.Limits.JetStreamLimits = limits2
+	aJwt2, err := claim.Encode(oKp)
+	require_NoError(t, err)
+	claim.Limits.JetStreamLimits = limitsExceeded
+	aJwtLimitsExceeded, err := claim.Encode(oKp)
+	require_NoError(t, err)
+	claim.Limits.JetStreamLimits = jwt.JetStreamLimits{} // disabled
+	aJwt4, err := claim.Encode(oKp)
+	require_NoError(t, err)
+	// account user
+	uKp, _ := nkeys.CreateUser()
+	uSeed, _ := uKp.Seed()
+	uclaim = newJWTTestUserClaims()
+	uclaim.Subject, _ = uKp.PublicKey()
+	userJwt, err := uclaim.Encode(akp)
+	require_NoError(t, err)
+	userCreds := genCredsFile(t, userJwt, uSeed)
+	dir, err := ioutil.TempDir("", "srv")
+	require_NoError(t, err)
+	defer os.RemoveAll(dir)
+	conf := createConfFile(t, []byte(fmt.Sprintf(`
+		listen: -1
+		jetstream: {max_mem_store: 10Mb, max_file_store: 10Mb}
+		operator: %s
+		resolver: {
+			type: full
+			dir: %s
+		}
+		system_account: %s
+    `, ojwt, dir, sysPub)))
+	defer os.Remove(conf)
+	opts := LoadConfig(conf)
+	s := RunServer(opts)
+	defer s.Shutdown()
+
+	updateJwt(s.ClientURL(), sysCreds, sysPub, sysJwt)
+	sys := natsConnect(t, s.ClientURL(), nats.UserCredentials(sysCreds))
+	expect_InfoError(sys)
+	sys.Close()
+	updateJwt(s.ClientURL(), sysCreds, aPub, aJwt1)
+	c := natsConnect(t, s.ClientURL(), nats.UserCredentials(userCreds), nats.ReconnectWait(200*time.Millisecond))
+	defer c.Close()
+	validate_limits(c, limits1)
+	// keep using the same connection
+	updateJwt(s.ClientURL(), sysCreds, aPub, aJwt2)
+	validate_limits(c, limits2)
+	// keep using the same connection but do NOT CHANGE anything.
+	// This tests if the jwt is applied a second time (would fail)
+	updateJwt(s.ClientURL(), sysCreds, aPub, aJwt2)
+	validate_limits(c, limits2)
+	// keep using the same connection. This update EXCEEDS LIMITS
+	updateJwt(s.ClientURL(), sysCreds, aPub, aJwtLimitsExceeded)
+	validate_limits(c, limits2)
+	// disable test after failure
+	updateJwt(s.ClientURL(), sysCreds, aPub, aJwt4)
+	expect_InfoError(c)
+	// re enable, again testing with a value that can't be applied twice
+	updateJwt(s.ClientURL(), sysCreds, aPub, aJwt2)
+	validate_limits(c, limits2)
+	// disable test no prior failure
+	updateJwt(s.ClientURL(), sysCreds, aPub, aJwt4)
+	expect_InfoError(c)
+	// Wrong limits form start
+	updateJwt(s.ClientURL(), sysCreds, aPub, aJwtLimitsExceeded)
+	expect_JSDisabledForAccount(c)
+	// enable js but exceed limits. Followed by fix via restart
+	updateJwt(s.ClientURL(), sysCreds, aPub, aJwt2)
+	validate_limits(c, limits2)
+	updateJwt(s.ClientURL(), sysCreds, aPub, aJwtLimitsExceeded)
+	validate_limits(c, limits2)
+
+	s.Shutdown()
+	conf = createConfFile(t, []byte(fmt.Sprintf(`
+		listen: %d
+		jetstream: {max_mem_store: 20Mb, max_file_store: 20Mb}
+		operator: %s
+		resolver: {
+			type: full
+			dir: %s
+		}
+		system_account: %s
+    `, opts.Port, ojwt, dir, sysPub)))
+	defer os.Remove(conf)
+	s = RunServer(LoadConfig(conf))
```

Comment:
Same than above. And if you don't need options, you can always do `s, _ = RunServerWithConfig(conf)`

Reply:
will change